### PR TITLE
Don't capture empty session tokens from env

### PR DESCRIPTION
**Pull request by jdisanti**
## Motivation and Context
This is the fix for https://github.com/awslabs/aws-sdk-rust/issues/316

The AWS CLI won't send session tokens if `AWS_SESSION_TOKEN` is empty or has nothing but whitespace.

## Testing
- Added a unit test
- Manually tested with S3 example

## Checklist
- [x] I have updated `aws/SDK_CHANGELOG.md` if I made changes to the AWS SDK, generated SDK code, or SDK runtime crates

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws/rust-runtime/aws-config/src/environment/credentials.rs [hunk 1]
```diff
@@ -29,7 +29,12 @@ impl EnvironmentVariableCredentialsProvider {
             .get("AWS_SECRET_ACCESS_KEY")
             .or_else(|_| self.env.get("SECRET_ACCESS_KEY"))
             .map_err(to_cred_error)?;
-        let session_token = self.env.get("AWS_SESSION_TOKEN").ok();
+        let session_token = self
+            .env
+            .get("AWS_SESSION_TOKEN")
+            .ok()
+            .map(|token| token.trim().to_string())
+            .filter(|token| !token.is_empty());
```
Comment (Velfi) [inline aws/rust-runtime/aws-config/src/environment/credentials.rs:37]: Is it worth it to refactor this so we allocate after the filter? We could even use `filter_map`
```
.filter_map(|token| {
  let t = token.trim();
  
  if token.is_empty() {
    None
  } else {
    Some(token.to_string())
  }
})
```

**Review by Velfi — APPROVED** [review]

**Review by rcoh — APPROVED** [review]
LGTM

File: aws/rust-runtime/aws-config/src/environment/credentials.rs (hunk 1 above)
Comment (jdisanti) [reply aws/rust-runtime/aws-config/src/environment/credentials.rs:37]: It doesn't look like [Option](https://doc.rust-lang.org/stable/std/option/enum.Option.html) has `filter_map`. Could potentially map it to another option and then `flatten` though.

Comment (Velfi) [reply aws/rust-runtime/aws-config/src/environment/credentials.rs:37]: I always forget I can't user iterator methods on options and results

**Pull request merged**